### PR TITLE
Update React Fundamental in Compass

### PR DESCRIPTION
**Pull request by Wawrzyn321**
**Description**

Changes proposed in this pull request:

- Update React Fundamental in Compass
  - As new FD `Popover` click events won't work in` Modal`s from `@kyma-project/react-components`, create new custom `Modal`.
We could use original FD `Modal`, but it would introduce more breaking changes to already existing code.

- Update affected code & tests accordingly, removing some "key prop" checks.

**Related issue(s)**

[Kyma](https://github.com/kyma-project/kyma/pull/6494)
[Compass](https://github.com/kyma-incubator/compass/pull/527)

### CODE REVIEW

**Comment by Wawrzyn321** [comment]
Most of the `key prop` errors come from `GenericList`, which still uses non-upgraded Fundamental version.

File: compass/src/components/Application/ApplicationDetails/ConnectApplicationModal/test/ConnectApplicationModal.test.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+    await openModal(component);
+
+    const inputs = component
+      .findWhere(t => t.type() === 'input')
+      .map(i => i.instance().value);
+    expect(inputs).toMatchSnapshot();
+  });
+
+  // Timeout - Async callback was not invoked within the 5000ms timeout specified by jest.setTimeout.Timeout
```
Comment (dariadomagala) [inline compass/src/components/Application/ApplicationDetails/ConnectApplicationModal/test/ConnectApplicationModal.test.js]: Why is it commented out?

File: compass/src/setupTests.js [hunk 2]
```diff
@@ -14,4 +14,14 @@ global.wait = require('waait');
 
 window.postMessage = jest.fn();
 
+// fix UnhandledPromiseRejectionWarning: TypeError: document.createRange is not a function
```
Comment (dariadomagala) [inline compass/src/setupTests.js]: Where does this error come from?

**Pull request merged**